SimpleWeather/DailyWeatherForecast/DailyWeatherForecastInteractor.swift

```diff
@@ -0,0 +1,121 @@
+         modelConverter: WeatherForecastConverter,
+         databaseManager: DatabaseManager,
+         currentUserSession: CurrentUserSession) {
+        self.httpClient = httpClient
+        self.modelConverter = modelConverter
+        self.databaseManager = databaseManager
+        self.currentUserSession = currentUserSession
+    }
+    
+    func retreiveDailyWeatherForecast() {
+        self.locationManager.getCurrentLocation()
+    }
+    
+    func retreiveCurrentDailyWeatherForecast(_ locations: [CLLocation]?, isCurrent: Bool) {
+        
+        guard
+            let lat = locations?.first?.coordinate.latitude,
+            let lon = locations?.first?.coordinate.longitude
+        else { return }
+        
+        httpClient.fetchCurrentWeather(
+            parameters: ["lat": lat,
+                         "lon": lon,
+                         "units": "metric"],
+            completionHandler: { dailyWeatherResult in
+                switch dailyWeatherResult {
+                case .success(let dailyWeatherResponse):
+                    self.httpClient.fetchCurrentHourlyWeather(
+                        parameters: ["lat": lat,
+                                     "lon": lon,
+                                     "units": "metric"],
+                        completionHandler: { dailyWeeklyHourlyResult in
+                            switch dailyWeeklyHourlyResult {
+                            case .success(let dailyWeeklyHourlyResponse):
+                                let weatherForecast =
+                                    self.modelConverter.convertWeatherForecast(dailyWeatherResponse,
+                                                                               dailyWeeklyHourlyResponse)
+                                self.didRetreieveWeatherForecastFromNetwork(weatherForecast)
+                                self.saveCitiesToDatabase(weatherForecast.cityName,
+                                                          lattitude: Float(lat),
+                                                          longitude: Float(lon),
+                                                          isCurrent: isCurrent)
+                                self.isComplete = false
+                            case .failure(let err):
+                                print(err)
+                            }
+                    })
+                case .failure(let err):
+                    print(err)
+                }
+        })
+    }
+    
+    func saveCitiesToDatabase(_ weatherCityName: String,
+                              lattitude: Float,
+                              longitude: Float,
+                              isCurrent: Bool) {
+        databaseManager.saveCity(weatherCityName,
+                                 lattitude: lattitude,
+                                 longtitude: longitude, isCurrent: isCurrent)
+    }
+    
+    func didRetreieveWeatherForecastFromNetwork(_ weatherForecast: WeatherForecast?) {
```

Comment:
у тебя сигнатуре метода говорит что ты его получил, но принимаешь ты зачем то нил, уберешь иф лет снизу за одно
```suggestion
    func didRetreieveWeatherForecastFromNetwork(_ weatherForecast: WeatherForecast) {
```